tests/integration/targets/ec2_metadata_facts/playbooks/setup.yml

```diff
@@ -115,8 +115,11 @@
       network:
         assign_public_ip: true
         delete_on_termination: true
+      metadata_options:
```

Comment:
The instance type is wrong since you need Nitro support (https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html#ec2-nitro-instances). I personally don't care if there is functional test or not since the code follow the same logic than the rest of the facts (assuming here you just use `self.fetch()`).